clap_derive/tests/argument_naming.rs

```diff
@@ -99,23 +99,24 @@ fn test_standalone_short_generates_kebab_case() {
 fn test_custom_short_overwrites_default_name() {
     #[derive(Clap, Debug, PartialEq)]
     struct Opt {
-        #[clap(short = "o")]
+        #[clap(short = 'o')]
         foo_option: bool,
     }
 
     assert_eq!(Opt { foo_option: true }, Opt::parse_from(&["test", "-o"]));
 }
 
-#[test]
-fn test_standalone_short_uses_previous_defined_custom_name() {
-    #[derive(Clap, Debug, PartialEq)]
-    struct Opt {
-        #[clap(name = "option", short)]
-        foo_option: bool,
-    }
-
-    assert_eq!(Opt { foo_option: true }, Opt::parse_from(&["test", "-o"]));
-}
+// TODO: Fix this situation
```

Comment:
I wasn't sure how to get this use case working. I'm relatively new to Rust macros, so I commented it out and hoped I good get some help on it!

Reply:
It's not `name` expecting a character, it's `short` expecting a character but getting `&str`. You're just confused because the error points to the value instead of the method.

```rust
    pub fn translate_char(self, style: CasingStyle) -> TokenStream {
        use CasingStyle::*;

        match self {
            /// `tokens` here is `&str`, not `char`. We need to convert it to char at runtime.
            Name::Assigned(tokens) => quote!( (tokens).chars().next().unwrap() ),
```